src/model_diagnostics/xai/permutation_importance.py

```diff
@@ -0,0 +1,292 @@
+import copy
+from typing import Callable, Optional, Union
+
+import numpy as np
+import numpy.typing as npt
+import polars as pl
+
+from model_diagnostics._utils.array import (
+    get_second_dimension,
+    is_pandas_df,
+    is_pyarrow_table,
+    length_of_first_dimension,
+    safe_assign_column,
+    safe_index_rows,
+)
+from model_diagnostics.scoring import SquaredError
+
+
+def safe_copy(X):
```

Comment:
Might be good to put `safe_copy` and `safe_column_names` into `_utils.array` and add tests for them. I think they cause the current CI failure.

Reply:
My local tests are failing for the Python 3.9 environment only (pandas and pyarrow). I will move the functions to _utils.array, draft some unit tests, and rename `safe_column_names()` to `get_column_names()`.